src/test/java/com/pnu/dev/shpaltaif/PublicAccountServiceIntegrationTest.java

```diff
@@ -128,4 +133,65 @@ public void createAndThenReadAndThenUpdateAndThenDeleteSuccessFlow() {
         assertEquals(0, publicAccountsAfterDelete.size());
     }
 
+    @Test
+    @Transactional
+    public void updatePseudonymTest() {
+
+        PublicAccount publicAccount1 = createUserWriter("username1").getPublicAccount();
+
+        PublicAccountDto publicAccountDtoWithoutPseudonym = PublicAccountDto.builder()
+                .name(publicAccount1.getName())
+                .surname(publicAccount1.getSurname())
+                .pseudonymUsed(true)
+                .build();
+
+        ServiceException thrown = assertThrows(ServiceException.class,
+                () -> publicAccountService.update(publicAccountDtoWithoutPseudonym, publicAccount1.getId()));
+        assertEquals(thrown.getMessage(), "Щоб використовувати псевдонім, введіть його коректно");
+
+        PublicAccountDto publicAccountDtoWithPseudonym = PublicAccountDto.builder()
+                .name(publicAccount1.getName())
+                .surname(publicAccount1.getSurname())
+                .pseudonymUsed(true)
+                .pseudonym(" Pseudonym ")
+                .build();
+
+        publicAccountService.update(publicAccountDtoWithPseudonym, publicAccount1.getId());
+        PublicAccount updatedPublicAccount = publicAccountService.findById(publicAccount1.getId());
+        assertEquals(updatedPublicAccount.getPseudonym(), publicAccountDtoWithPseudonym.getPseudonym().trim());
+        assertEquals(updatedPublicAccount.isPseudonymUsed(), publicAccountDtoWithPseudonym.isPseudonymUsed());
+
+        PublicAccount publicAccount2 = createUserWriter("username2").getPublicAccount();
+        PublicAccountDto publicAccountDtoWithExistsPseudonym = PublicAccountDto.builder()
+                .name(publicAccount2.getName())
+                .surname(publicAccount2.getSurname())
+                .pseudonymUsed(true)
+                .pseudonym("Pseudonym")
+                .build();
+
+        thrown = assertThrows(ServiceException.class,
+                () -> publicAccountService.update(publicAccountDtoWithExistsPseudonym, publicAccount2.getId()));
+        assertEquals(thrown.getMessage(), "Псевдонім зайнятий");
+    }
+
+    private User createUserWriter(String username) {
```

Comment:
I'd suggest testing the following flow instead:
1. Create and save user and account (I'd use userRepository for dealing with user).
2. Check that publicAccount.getSignature() is what you expect.
3. Update account and check publicAccount.getSignature() again.

And another test for incorrect signature and duplicate username.

I'd recommend using userRepository instead of userService, because here you use side effect of user creation, that creates empty account, but actually it's the piece of logic that have to be tested by tests for user creation flow.